### ISSUE TITLE
Unable to build NuGet.Core.sln (with ',' as decimal separator)

### ISSUE THREAD

**Issue by forderud** [NuGet/Home#2179]
Steps to reproduce:
- Clone NuGet.Client repo & switch to latest commit on "dev" branch (1756e1e0dbc5ef56435d1b80b6cc1d2465496142 as of 2016-02-25).
- Open NuGet.Core.sln in VS2015 community, Update 1.
- Build NuGet.Common in debug from IDE.

Error message:

```
1>------ Build started: Project: NuGet.Common, Configuration: Debug Any CPU ------
1>  Setting buid environment variables to #2524708082214...
1>  C:\Users\Fredrik\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\dnx.exe --appbase "C:\Views\NuGet\NuGet.Client\src\NuGet.Core\NuGet.Common" "C:\Users\Fredrik\.dnx\runtimes\dnx-clr-win-x86.1.0.0-rc1-update1\bin\lib\Microsoft.Dnx.Tooling\Microsoft.Dnx.Tooling.dll" pack "C:\Views\NuGet\NuGet.Client\src\NuGet.Core\NuGet.Common" --configuration Debug --out "C:\Views\NuGet\NuGet.Client\artifacts\NuGet.Common"
1>  Microsoft .NET Development Utility Clr-x86-1.0.0-rc1-16231
1>DNX : error : Value was either too large or too small for an Int32.
========== Build: 0 succeeded or up-to-date, 1 failed, 0 skipped ==========
```

32 out of 55 projects in the solution fail with the same erorr message. Building with the build.ps1 script and NuGet.Client.sln succeeds, so the problem is limited to NuGet.Core.sln

**Comment by yishaigalatzer** [comment]
You have to build once with the build.ps1 script first, does it still fail afterwards?

**Comment by forderud** [comment]
Building first with build.ps1 does not seem to help.

**Comment by yishaigalatzer** [comment]
@forderud I do see a bunch of folks being able to figure this out, are you unblocked here?

@deepakaravindr please shepard this along

**Comment by forderud** [comment]
My problem still persists, even after pulling in the latest commits from the "dev" branch.

Could the problem be related to me using the "community" version of VS2015, and not the "professional" version? Alternatively, do I need the not yet released VS2015 update 2?

**Comment by TimBarham** [comment]
@forderud - just as a data point, my home machine is also on VS2015 Community, with update 1, and it is building fine there for me.

The problem seems to be your build number:

```
Setting buid environment variables to #2524708082214...
```

(Heh, seems we have a typo there that needs to be fixed)

That is certainly too large to fit in an `Int32`. It looks like that value is `BuildNumber`, calculated in `build\common.xproj.props` - it takes the number of minutes since 2015-11-30 and divides by 5 (that jibes with the value I'm saw with my last build - 28795).

**Comment by forderud** [comment]
Interesting.. Then I'm guessing that different regional settings might be the reason for why the build only fails for me.

**Comment by forderud** [comment]
My problem can be reproduced by changing the decimal separator to ',' in the Windows region settings.

This causes the `$([System.TimeSpan]::FromTicks($(SpanTicks)).TotalMinutes)` expression (in build\common.xproj.props) to return a floating point string with a ',' as decimal separator. This in turn is parsed incorrectly in the next line.

**Comment by yishaigalatzer** [comment]
Thanks!

**Comment by deepakaravindr** [comment]
Looks like this bug is fixed with https://github.com/NuGet/NuGet.Client/pull/356. Closing it

**Comment by zhili1208** [comment]
Verified fixed with 3.5.0-beta (build 3.5.0.1249)
